Async-Context-Definitions.md

```diff
@@ -302,48 +310,61 @@ The events described above can be used to construct and maintain an
+  * An "execution context" node is defined when an "executeBegin" event 
+  is received and all linking/causal events will create child edges from 
+  this node until the matching "executeEnd" event is seen.
 
+  * A "linking context" node is defined when a "link" event is received and 
+  is associated with the data from the link event.
 
-Edges are defined as follows: 
-  * A `link edge`, directed from a given "execution context" node 
-  _c<sub>1</sub>_ to second "linking context" node, _c<sub>2</sub>_
-  exists, if the `asynchronous event trace` contains the entries
-  _{event: "executeBegin", current: c<sub>1</sub>, time: t}_ and 
-  _{event: "link", linkCtx: c<sub>2</sub>, time: t'}_ where t < t' and, 
-  if the trace contains an event 
-  _{event: "executeEnd", current: c<sub>1</sub>, time: t''}_, where t < t'' then t' < t''.
-  In this case, a `link edge` exists directed from _c<sub>1</sub>_ to _c<sub>2</sub>_.
-
-  * A `causal edge`, directed from a given "execution context" node, _c<sub>1</sub>_,
-  to a 
-  second "execution context" node, _c<sub>2</sub>_, if the `asynchronous event 
-  trace` contains the entries
-  _{event: "executeBegin", current: c<sub>1</sub>, time: t}_ and 
-  _{event: "cause", causeCtx: c<sub>2</sub>, time: t'}_ where t < t' and, 
-  if the trace contains an event 
-  _{event: "executeEnd", current: c<sub>1</sub>, time: t''}_, where t < t'' then t' < t''.
- 
-  * An `execution edge`, directed from a given "linking context" node, _c<sub>1</sub>_, 
-  to a second `linking context` node _c<sub>2</sub>_, if the `asynchronous event trace` 
-    contains the entries
-  _{event: "link", linkCtx: c<sub>1</sub>, time: t}_ and 
-  _{event: "executeBegin", current: c<sub>2</sub>, time: t''}_, where t < t'' then t' < t''.
+  * A "causal context" node is defined when a "cause" event is received and 
+  is associated with the data from the cause event.
 
-  **TODO** - need better definition of above - insuficient info in the event trace to know 
-  that c<sub>1</sub> is the linking node for c<sub>2</sub> 
+Edges are defined as follows: 
+  * A "link" edge is added from the current "execution context" node 
+  to a newly created "linking context" node when a "link" event is 
+  received. 
 
+  * A "causal" edge is added from the current "execution context" node 
+  to a newly created "causal context" node when a "cause" event is 
+  received. An additional "linked by" cross-edge is added to the 
+  "linking context" node with the corresponding link context value.
 
+  * An "execution" edge is added into the newly created "execution context" 
+  node from the "causual context" node with the corresponding causal context 
+  value.
 
 **TODO** use these definitions to see trees for trace examples.
 
-### Long Call-Stacks from Traces and Trees
-**TODO** fill this in
-
 ### Resource Use Monitoring with Traces
 **TODO** fill this in
 
+### Long Call-Stacks from Graphs

```

Comment:
I was thinking that stuff like this would be in section 5.  i.e., precisely how do you solve a given problem (long stacks, CLS, async exception handling,...) given the previous definitions.